openedx/tutor-contrib-aspects
# feat: allow to host dbt docs on cluster

### Description

This PR allows hosting dbt docs locally and on k8s. It also stores the `manifest.json` state file and superset exposures in clickhouse enabling us to remove the volume for the state. Closes: https://github.com/openedx/tutor-contrib-aspects/issues/915

- Fixes an issue with the dbt state that makes it find false positives.
- Turn every rendered value into an env var.
- Removes the k8s volume for the dbt state in favor of storing metadata in ClickHouse.

### Testing instructions

- Enable `RUN_ASPECTS_DOCS` in your `config.yml`.
- Run migrations with `tutor dev do alembic -c "upgrade head"`
- Build the aspects docker image: `tutor images build aspects`
- Run DBT.
- Run collect lineage command.
- Go to http://dbt.local.edly.io:7000 and verify the docs are working, lineage data is available and exposures are available.
- Rerun DBT and verify that no changes have been detected.
